### PR TITLE
doc: release: Add Zephyr 3.6 highlights

### PR DESCRIPTION
**Pull request by kartben**
Taking a first stab at adding a high-level summary of the most notable highlights of Zephyr 3.6.

Maybe we can use this PR as a way to discuss & converge towards the final list, esp. as more release notes keep coming in (pretty sure -and hopeful!- there should be at least one entry related to Bluetooth in there)?
WDYT @henrikbrixandersen @MaureenHelm & others?

Edit: "as more release notes keep coming in "... and as I make progress on the announcement blog post, as I'm sure I will be spotting more, that at least to me will be worth highlighting :)

### CODE REVIEW

**Review by henrikbrixandersen — COMMENTED** [review]
This is a great start, thanks! Perhaps we could add links to the relevant documentation for each entry?

**Comment by kartben** [comment]
> This is a great start, thanks! Perhaps we could add links to the relevant documentation for each entry?

Yes I was thinking the same, we should certainly do that for sure. And if an item doesn't have a "good enough" link for us to use, now would probably be a good time to make sure there is one, as we want the highlights to be things people can easily learn more about.

**Comment by kartben** [comment]
> Love it! I am working on a section in the release notes for GNSS as well :)

Eheh, excellent! If you're bored, I would love to have a demo of cellular + gnss + lvgl, i.e. get GPS coordinates, compute the corresponding openstreetmap tile URL, fetch it and display it on the screen :)
I will try to get to it but I might simply run out of time. Should literally be just about 50 lines of code I'm sure, and would be VERY COOL (and a great sample/demo to actually add upstream...). Bonus points if we hook up some of @fabiobaltieri's input magic to zoom in / zoom out when pressing key on a keyboard :-)

**Comment by fabiobaltieri** [comment]
@kartben wanna mention something about the keyboard matrix driver? It's just a driver in a sea of drivers but it'd be nice to mention since it it does not depend on anything in particular and could be used with any platform we support.

**Comment by kartben** [comment]
> @kartben wanna mention something about the keyboard matrix driver? It's just a driver in a sea of drivers but it'd be nice to mention since it it does not depend on anything in particular and could be used with any platform we support. (but also happy with not mentioning on the basis of: it's just a driver)

Just read again :-) Guess the wording of that particular bullet item might be backwards or something, hence why you maybe missed it. Please suggest a better wording if you can think of one!
And yes, "just" a driver but worth calling out imo

**Comment by bjarki-trackunit** [comment]
@kartben I'm game helping you make the sample work!

**Comment by fabiobaltieri** [comment]
@kartben sorry somehow I missed it :-) plenty good thanks for the mention

**Comment by MaureenHelm** [comment]
> > This is a great start, thanks! Perhaps we could add links to the relevant documentation for each entry?
> 
> Yes I was thinking the same, we should certainly do that for sure. And if an item doesn't have a "good enough" link for us to use, now would probably be a good time to make sure there is one, as we want the highlights to be things people can easily learn more about.

This is great, thanks @kartben . Like the idea to add docs links.

Should add a bullet about #60063. @yperess @teburd

**Comment by yperess** [comment]
> > > This is a great start, thanks! Perhaps we could add links to the relevant documentation for each entry?
> > 
> > Yes I was thinking the same, we should certainly do that for sure. And if an item doesn't have a "good enough" link for us to use, now would probably be a good time to make sure there is one, as we want the highlights to be things people can easily learn more about.
> 
> This is great, thanks @kartben . Like the idea to add docs links.
> 
> Should add a bullet about #60063. @yperess @teburd 

I think it's worth a mention, I can write something up tonight if that helps and post it here as a comment.

**Comment by kartben** [comment]
> I think it's worth a mention, I can write something up tonight if that helps and post it here as a comment.

That would be very helpful yes, and yes I agree it's totally worth mentioning. One challenge though is that it doesn't look like we would have something (think: doc page or code sample) to point people to?

If you are to write something tonight, I'd rather this would be some blurb to maybe add to the sensor shell sample explaining how one may play with the streaming API, as this would make a huge difference in the "discoverability" of this feature, including for all the folks who might not even read the release notes and could easily miss it for it's not really documented (i.e searching "streaming" gives virtually no result)
Thanks!

**Comment by yperess** [comment]
> 
> > I think it's worth a mention, I can write something up tonight if that helps and post it here as a comment.
> 
> That would be very helpful yes, and yes I agree it's totally worth mentioning. One challenge though is that it doesn't look like we would have something (think: doc page or code sample) to point people to?
> 
> If you are to write something tonight, I'd rather this would be some blurb to maybe add to the sensor shell sample explaining how one may play with the streaming API, as this would make a huge difference in the "discoverability" of this feature, including for all the folks who might not even read the release notes and could easily miss it for it's not really documented (i.e searching "streaming" gives virtually no result)
> Thanks!

We were going to hold off on the docs until we had better rtio support which @teburd is currently working on. So we'll probably have a big pay for documenting and advertising going into 3.7

**Comment by yperess** [comment]
@kartben sorry for the delay, @teburd any changes that you'd like to make?

> Introduced a new async sensor API that's 100% backwards compatible with the previous fetch/get blocking calls. It is
> possible to enable the API via the `CONFIG_SENSOR_ASYNC_API`. This API should work on any sensor driver that exists
> in Zephyr currently. On-top of this API, an additional streaming API was added. When enabled, the API works similarly
> to the trigger API but provides a user thread safe way to get the data. Additionally, the streaming API provides 3
> different ways to handle the trigger:
> 1. `include` - where the data produced by the sensor is included with the completion queue entry
> 2. `nop` - only notify the completion queue of the event, but do nothing with the data
> 3. `drop` - notify the completion queue of the event, and clear out the data (example: FIFO flush)
>
> Related APIs: RTIO

**Comment by teburd** [comment]
> > > I think it's worth a mention, I can write something up tonight if that helps and post it here as a comment.
> > 
> > 
> > That would be very helpful yes, and yes I agree it's totally worth mentioning. One challenge though is that it doesn't look like we would have something (think: doc page or code sample) to point people to?
> > If you are to write something tonight, I'd rather this would be some blurb to maybe add to the sensor shell sample explaining how one may play with the streaming API, as this would make a huge difference in the "discoverability" of this feature, including for all the folks who might not even read the release notes and could easily miss it for it's not really documented (i.e searching "streaming" gives virtually no result)
> > Thanks!
> 
> We were going to hold off on the docs until we had better rtio support which @teburd is currently working on. So we'll probably have a big pay for documenting and advertising going into 3.7

I agree this is the time frame when I'd really like to showcase and advertise these things more

**Comment by aescolar** [comment]
> (pretty sure -and hopeful!- there should be at least one entry related to Bluetooth in there)?

@jori-nordic @jhedberg @Thalley @cvinayak @PavelVPV ^^
Anything BT related we should add to the highlights section of the release notes?

**Comment by Thalley** [comment]
> > (pretty sure -and hopeful!- there should be at least one entry related to Bluetooth in there)?
> 
> @jori-nordic @jhedberg @Thalley @cvinayak @PavelVPV ^^ Anything BT related we should add to the highlights section of the release notes?

I don't have anything specific

**Comment by PavelVPV** [comment]
> > (pretty sure -and hopeful!- there should be at least one entry related to Bluetooth in there)?
> 
> @jori-nordic @jhedberg @Thalley @cvinayak @PavelVPV ^^ Anything BT related we should add to the highlights section of the release notes?

Could this line go into highlights:
https://github.com/zephyrproject-rtos/zephyr/blob/c33743ef955b066a5711097a6b0e184de1519b7f/doc/releases/release-notes-3.6.rst#L105

**Comment by kartben** [comment]
> > > (pretty sure -and hopeful!- there should be at least one entry related to Bluetooth in there)?
> > 
> > 
> > @jori-nordic @jhedberg @Thalley @cvinayak @PavelVPV ^^ Anything BT related we should add to the highlights section of the release notes?
> 
> Could this line go into highlights?
> 
> https://github.com/zephyrproject-rtos/zephyr/blob/c33743ef955b066a5711097a6b0e184de1519b7f/doc/releases/release-notes-3.6.rst#L105

added in the latest push

**Comment by jori-nordic** [comment]
@aescolar should we mention the UART you made + the fact that HCI H4 sample + driver combo is now tested in CI?
Maybe not in the highlights, but I definitely think it warrants a mention somewhere. A lot of people depend on it and it was previously not tested at all.

**Comment by aescolar** [comment]
> @aescolar should we mention the UART you made + the fact that HCI H4 sample + driver combo is now tested in CI? 

Thanks @jori-nordic . It is mentioned at the end on the testing & samples section

**Review by MaureenHelm — DISMISSED** [review]

**Review by aescolar — DISMISSED** [review]

**Review by henrikbrixandersen — DISMISSED** [review]

File: doc/releases/release-notes-3.6.rst [hunk 1]
```diff
@@ -9,6 +9,18 @@ We are pleased to announce the release of Zephyr version 3.6.0.
 
 Major enhancements with this release include:
 
+* New :ref:`GNSS subsystem <gnss_api>` added, enabling geo-awareness in Zephyr applications.
+* New API and drivers introduced for interfacing with `keyboard matrices <gpio-kbd>`.
```
Comment (fabiobaltieri) [inline doc/releases/release-notes-3.6.rst:13]: missing a `:ref:`? same for native_sim below

**Comment by kartben** [comment]
> @kartben Can we please remove the DNM label and get this merged so we can finalize the release notes?

yes! Added LTO and fixed the issues pointed by  @fabiobaltieri

**Review by jhedberg — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Review by bjarki-andreasen — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**